### PR TITLE
Update `combine_tensor_patches`

### PR DESCRIPTION
**Pull request by ashnair1**
This PR aims to have `combine_tensor_patches` expose a similar API as `extract_tensor_patches`

#### Changes
- Have `combine_tensor_patches` accept `int` or `Tuple[int,int]`
- Have unpadding also accept Tuple[int, int, int, int]
- Remove default value for args
- Bugfix for padding
- Add tests for invalid padding

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by ashnair1** [comment]
Note:
unpadding is tuple of 4 ints. Where the first two values are for the first dim (H) of image and last two values are for the second (W) dim of image

**Review by edgarriba — APPROVED** [review]
LGTM. Pleas, @shijianjian review and merge

**Comment by ashnair1** [comment]
Tests were failing due to mypy. Pushed fix

**Comment by shijianjian** [comment]
> * Should we enable padding to accept a tuple of 4 ints in `extract_tensor_patches` as well for consistency? Or should I revert unpadding in `combine_tensor_patches` to tuple of 2 ints? Adding a 4 tuple option might enable more flexibility but I'm not sure how common it is for user to set unequal padding in the same dimension.

My suggestion is to have as complicated but powerful functional functions, then add several syntax sugar for the `nn.Module`-like operations.

> ~- In this PR, I have done variable redefinitions which mypy by default doesn't like. If you use the `--allow-redefinition` option there isn't an issue. Is this fine or shall I go with assigning new variables?~

Better assign new variables here.

**Comment by ashnair1** [comment]
> > ~- In this PR, I have done variable redefinitions which mypy by default doesn't like. If you use the `--allow-redefinition` option there isn't an issue. Is this fine or shall I go with assigning new variables?~
> 
> Better assign new variables here.

I was able to get around this using `typing.cast`

**Comment by ashnair1** [comment]
> > * Should we enable padding to accept a tuple of 4 ints in `extract_tensor_patches` as well for consistency? Or should I revert unpadding in `combine_tensor_patches` to tuple of 2 ints? Adding a 4 tuple option might enable more flexibility but I'm not sure how common it is for user to set unequal padding in the same dimension.
> 
> My suggestion is to have as complicated but powerful functional functions, then add several syntax sugar for the `nn.Module`-like operations.

Sure. Shall I do it here or in a follow up PR?

**Comment by ashnair1** [comment]
> > > * Should we enable padding to accept a tuple of 4 ints in `extract_tensor_patches` as well for consistency? Or should I revert unpadding in `combine_tensor_patches` to tuple of 2 ints? Adding a 4 tuple option might enable more flexibility but I'm not sure how common it is for user to set unequal padding in the same dimension.
> > 
> > 
> > My suggestion is to have as complicated but powerful functional functions, then add several syntax sugar for the `nn.Module`-like operations.
> 
> Sure. Shall I do it here or in a follow up PR?

@shijianjian I went ahead and made the changes here. Think it's ready to go now

File: kornia/contrib/extract_patches.py [hunk 1]
```diff
@@ -1,10 +1,12 @@
-from typing import Optional, Tuple, Union
+from typing import Optional, Tuple, Union, cast
 
 import torch
 import torch.nn as nn
 import torch.nn.functional as F
 from torch.nn.modules.utils import _pair
 
+PadType = Union[Tuple[int, int], Tuple[int, int, int, int]]
```
Comment (edgarriba) [inline kornia/contrib/extract_patches.py:8]: just one side note here -- torchscript might not work well here for the `Union` type

Comment (ashnair1) [reply kornia/contrib/extract_patches.py:8]: Does this extend to function args? If not, I could just replace `PadType` with `Union[Tuple[int, int], Tuple[int, int, int, int]]`.

Comment (ashnair1) [reply kornia/contrib/extract_patches.py:8]: Seems Union support will be added in Pytorch 1.11.0. Ref: pytorch/pytorch#64234

Comment (ashnair1) [reply kornia/contrib/extract_patches.py:8]: I think we can address the torchscript issue in a later PR. This PR is mainly to unify APIs and ensure correct behaviour. From some initial tests, solving scripting issues would involve removing `Union` (which would reduce flexibility of the functions) and some other functions like `torch.is_tensor` which needs some discussion.

Comment (edgarriba) [reply kornia/contrib/extract_patches.py:8]: agree -- torchscript it's another discussion that has been bothering us for a while :)

Are you in our slack ? We can have an open discussion there and collect feedback from out developers community.

Comment (ashnair1) [reply kornia/contrib/extract_patches.py:8]: Sounds good. Recently joined the kornia slack so we can continue there
